server/controllers/transactions.js

```diff
@@ -8,22 +8,28 @@ const dec = require("../utils/dec.js");
 const GENESIS_PARENT_UUID = "00000000-0000-0000-0000-000000000000";
 
 const getOldestTransaction = async (walletUuid, tokenUuid) => {
-  // TODO modify this function to find earliest that is not 
-  // const transaction = await Transaction.findAll({
-  //   where: {
-  //     toAddress: walletUuid,
-  //     tokentype_uuid: tokenUuid
-  //   },
-  //   limit: 1,
-  //   order: [["updatedAt", "ASC"]]
-  // });
-  // return transaction[0];
+  const transactions = await Transaction.findAll({
+    where: {
+      toAddress: walletUuid,
+      tokentype_uuid: tokenUuid
```

Comment:
prob need to check the status here -- then just sort ASC and limit by 1 so u can get the oldest one

Reply:
I do not know what you mean by status. I implemented this and it works effectively doing the following:
1. Retrieve all transactions of a public key / TokenType UUID
2. Put the parents of all of these transactions in a set.
3. Iterate through the transactions from oldest to newest until we find one that is NOT a parent transaction, which means it is the oldest transaction that hasn't been passed on to another user.

Reply:
aren't we not going to have a status to determine if the challenge is still active or not (if it has been redeemed)

Reply:
an optimization to speed this up might be to store the ID for the challenge in every tx